### PR TITLE
WIP: Tighten up /usr/bin/iiab-hotspot-off and /usr/bin/iiab-hotspot-on just a bit, to clarify?

### PR DESCRIPTION
**Pull request by holta**
Just some preliminary ideas for discussion.  No significant/functional code changes, except for minor de-duplication of a couple lines.

In support of clarifying various "internal hotspot" gears/levers for IIAB operators who prefer CLI, e.g.:

- #3541

### CODE REVIEW

**Comment by jvonau** [comment]
You break it you bought it.

**Comment by holta** [comment]
FYI @EMG70 will try to test this PR and PR #3542 simultaneously, by running:

```
curl iiab.io/install.txt | bash -s 3542 3546
```

If he has time this week, he might test multiple different OS's, possibly including:

- Ubuntu (Desktop?) 23.04 in a laptop
- 64-bit Raspberry Pi OS (Lite or "with desktop" ?)
- Debian 11 "Bullseye"
- Debian 12 "Bookworm"
- Mint 21

**Comment by jvonau** [comment]
There are no functional improvement to test here, what is the point? If you address https://github.com/iiab/iiab/issues/3541#issuecomment-1518946509 then there is a real improvement otherwise this is just fluff and no substance.

File: roles/network/templates/hostapd/iiab-hotspot-on [hunk 1]
```diff
@@ -9,18 +9,17 @@ echo -e "If you add Wi-Fi hardware, run 'cd /opt/iiab/iiab' then 'sudo ./iiab-ne
...
+#exit 0
 {% else %}
 {% if is_raspbian %}
 # just do what we have always done in hotspot-on
 cp -f /etc/hostapd/hostapd.conf.iiab /etc/hostapd/hostapd.conf
-sed -i -e "s/^#denyinterfaces/denyinterfaces/" /etc/dhcpcd.conf
+sed -i "s/^#denyinterfaces/denyinterfaces/" /etc/dhcpcd.conf
 # shut down wlan0 in case connected to network
 ip link set wlan0 down
 systemctl enable hostapd
```
Comment (holta) [inline roles/network/templates/hostapd/iiab-hotspot-on:25]: @jvonau 3 lines below this (on Line 29) `systemctl start hostapd` is run (if OS is RasPiOS).

Is there any hope for doing the same (i.e. running `systemctl start hostapd` within iiab-hotspot-on) on other OS's too — i.e. just below Line 44?

Separately: Should similar possibly be considered around Line 16 (which applies to all OS's) in the very separate case where `wifi_up_down` ?

Comment (jvonau) [reply roles/network/templates/hostapd/iiab-hotspot-on:25]: In short no, and the way that is written one can't use wifi_up_down False and install over wifi, the original reason for hotspot-on|off to begin with. You want to wear the chef's hat go ahead but I'm not your dishwasher.

Comment (holta) [reply roles/network/templates/hostapd/iiab-hotspot-on:25]: > In short no; the way that is written one can't use wifi_up_down False and install over wifi, the original reason for hotspot-on|off to begin with.

Good to know.  Thanks for clarifying.

**Comment by holta** [comment]
Conversation moved to:

- PR #3549

**Comment by jvonau** [comment]
close this one if 3549 is on the table

**Pull request closed without merging**